Source/WebGPU/WebGPU/Texture.mm

```diff
@@ -2157,10 +2157,11 @@ static bool textureViewFormatCompatible(WGPUTextureFormat format1, WGPUTextureFo
         return 0;
     }
     NSUInteger blocksInWidth = textureWidth / blockWidth;
-    return Texture::texelBlockSize(format) * blocksInWidth * sampleCount;
+    auto product = checkedProduct<NSUInteger>(Texture::texelBlockSize(format), blocksInWidth, sampleCount);
```

Comment:
I wonder if `Texture::texelBlockWidth()` and all the data members of `m_device->limits()` should changed to return Checked<T>?

I also wonder if you want to use `Checked<T, RecordOverflow>`, so that instead of writing `checkedProduct(a, b)` you can just write `a * b`?

If typing `RecordOverflow` everywhere is annoying, you can always add more convenience typedefs to CheckedArithmetic.h, like CheckedNSUInteger. (CheckedUint32 is already there.)